src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java

```diff
@@ -79,4 +127,106 @@ public UserTenantCollection getByUsernameAndTenantId(UUID consortiumId, String u
     result.setTotalRecords(1);
     return result;
   }
+
+  private User prepareShadowUser(UUID userId, UserTenantEntity userTenantEntity, FolioExecutionContext folioExecutionContext) {
+    try (var context = new FolioExecutionContextSetter(prepareContextForTenant(userTenantEntity.getTenant().getId(), folioExecutionContext))) {
+      User user = new User();
+      User userOptional = getUser(userId);
+
+      if (Objects.nonNull(userOptional.getId())) {
+        user.setId(UUID.randomUUID().toString());
+        user.setPatronGroup(PATRON_GROUP);
+        user.setUsername(userOptional.getUsername() + HelperUtils.randomString(RANDOM_STRING_COUNT));
+        var userPersonal = userOptional.getPersonal();
+        if (Objects.nonNull(userPersonal)) {
+          Personal personal = new Personal();
+          personal.setLastName(userPersonal.getLastName());
+          personal.setFirstName(userPersonal.getFirstName());
+          personal.setEmail(userPersonal.getEmail());
+          personal.setPreferredContactTypeId(userPersonal.getPreferredContactTypeId());
+          user.setPersonal(personal);
+        }
+        user.setPatronGroup(userOptional.getPatronGroup());
```

Comment:
I believe tenants do not share the same ids for patron groups. So I am estimating that some calls to create might fail. Have we had successful shadow user creation with patron id's set?